### PR TITLE
Add chart releaser mechanism [PLEN-200]

### PR DESCRIPTION
**Pull request by adamma-da**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,28 +3,38 @@
...
-  build:
+executors:
+  chart-releaser:
     docker:
-    - image: cimg/base:2021.04
+       - image: quay.io/helmpack/chart-releaser:v1.5.0
     environment:
-      <<: *env
+      LC_ALL: "C.UTF-8"
+    resource_class: medium
```
Comment (nicu-da) [inline .circleci/config.yml:12]: We can probably downgrade to small

File: .circleci/config.yml [hunk 2]
```diff
@@ -3,28 +3,38 @@
...
+
+commands:
+  build-cr:
     steps:
-      - checkout
       - run:
-         name: build
-         command: |
-           set -euo pipefail
+          command: apk add helm yq github-cli git jq openssh bash
```
Comment (nicu-da) [inline .circleci/config.yml:18]: Can we add a nix setup here to avoid running the apk add every time?

Comment (adamma-da) [reply .circleci/config.yml:18]: I tried this out here: https://app.circleci.com/pipelines/github/adamma-da/test-chart-releaser/192/workflows/96ebd360-39c1-4581-8606-f834a4fdc5b8/jobs/156
It seems to work without any issues although it makes the build time quite long. To keep this MR clean I would prepare another one as we agreed

File: .circleci/config.yml [hunk 3]
```diff
@@ -3,28 +3,38 @@
...
+      - checkout
+      - deploy-cr
 
 workflows:
-  version: 2
-  build:
+  releaser:
     jobs:
-    - build
+      - release-chart:
```
Comment (nicu-da) [inline .circleci/config.yml:37]: Shouldn't this also require an approval job? Or do we release on each main commit?

File: .circleci/config.yml (hunk 2 above)
Comment (davidd-da) [reply .circleci/config.yml:18]: We need to commit a Docker image, same issue as in our other pipelines... Nix will install packages too 😞

File: charts/canton-domain/Chart.yaml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ---
 apiVersion: v2
 name: "canton-domain"
-version: 0.0.6
+version: 0.1.0
```
Comment (davidd-da) [inline charts/canton-domain/Chart.yaml:4]: Let's bump the chart version in the private repository when we are ready, this will unsync things

Comment (adamma-da) [reply charts/canton-domain/Chart.yaml:4]: PR for that is here: https://github.com/DACH-NY/daml-enterprise-k8s-support/pull/96/files

File: .circleci/config.yml (hunk 3 above)
Comment (adamma-da) [reply .circleci/config.yml:39]: Please refresh and check again @nicu-da  :)

File: .circleci/config.yml [hunk 5]
```diff
@@ -34,7 +34,11 @@ jobs:
 workflows:
   releaser:
     jobs:
+      - hold:
```
Comment (nicu-da) [inline .circleci/config.yml:37]: Thinking about it, this job is an approval, therefore the name should probably indicate what we approve. maybe `approve_release_helm_charts`?

Comment (adamma-da) [reply .circleci/config.yml:37]: yes, that naming makes much more sense. Added

**Pull request merged**